shim/handler.go

```diff
@@ -544,35 +544,22 @@ func (h *Handler) sendBatch(channelID string, txid string) error {
 	return nil
 }
 
-func (h *Handler) handleStartWriteBatch(channelID string, txID string) error {
+func (h *Handler) handleStartWriteBatch(channelID string, txID string) {
 	if !h.usePeerWriteBatch {
-		return errors.New("peer does not support write batch")
+		return
```

Comment:
I guess the intent here is that chaincode developer can call StartWriteBatch(), and it will work regardless of whether peer supports it or not.
If supported, requests will be batched.
If not supported, requests will be sent individually (consistent with existing behavior).

That seems like the right approach to allow for hybrid environments where some peers may support it, while other peers do not yet support it.

Did I get that right?

Reply:
It might be worth clarifying that behavior in the godoc for StartWriteBatch.

I notice that the existing godoc for StartWriteBatch reads:

> StartWriteBatch enables a mode where all changes are not immediately forwarded to the **feast**

Perhaps **feast** was supposed to read **peer**?

Reply:
> I guess the intent here is that chaincode developer can call StartWriteBatch(), and it will work regardless of whether peer supports it or not. If supported, requests will be batched. If not supported, requests will be sent individually (consistent with existing behavior).
> 
> That seems like the right approach to allow for hybrid environments where some peers may support it, while other peers do not yet support it.
> 
> Did I get that right?

Yeah. That's right.

Reply:
> It might be worth clarifying that behavior in the godoc for StartWriteBatch.
> 
> I notice that the existing godoc for StartWriteBatch reads:
> 
> > StartWriteBatch enables a mode where all changes are not immediately forwarded to the **feast**
> 
> Perhaps **feast** was supposed to read **peer**?

That's right. Thank you for finding out. I fixed it.